File: packages/capp-sdk-ts/src/modules/wallet.ts
import { CAPPClient } from '../client';
import { Balances } from '../models';
import { handleApiError } from '../utils';

export class WalletModule {
    constructor(private client: CAPPClient) { }

    public async getBalance(params?: { chain?: string; currency?: string }): Promise<Balances> {
        const query = new URLSearchParams();
        if (params?.chain) query.append('chain', params.chain);
        if (params?.currency) query.append('currency', params.currency);

        const qs = query.toString();
        const url = `/wallet/balances${qs ? '?' + qs : ''}`;

        const res = await this.client.fetch(url);
        await handleApiError(res);
        return res.json();
    }
}

File: packages/capp-sdk/capp/modules/wallet.py
import httpx
from typing import Optional
from ..models import Balances
from .._utils import handle_api_error

class WalletModule:
    def __init__(self, client: httpx.AsyncClient):
        self._client = client
        
    async def get_balance(
        self, 
        chain: Optional[str] = None, 
        currency: Optional[str] = None
    ) -> Balances:
        params = {}
        if chain:
            params["chain"] = chain
        if currency:
            params["currency"] = currency
            
        res = await self._client.get("/wallet/balances", params=params)
        handle_api_error(res)
        return Balances(**res.json())
